apps/nextjs-website/src/components/PressReleaseList.tsx

```diff
@@ -14,28 +31,44 @@ const makePressReleaseListProps = ({
 }: PressReleaseListSection &
   SiteWidePageData & {
     pressReleasePages: ReadonlyArray<PressReleasePage>;
-  }): PressReleaseListProps => ({
-  ...rest,
-  pressReleases: pressReleasePages.map((PRPage) => ({
-    date: PRPage.pressRelease.date,
-    title: PRPage.pressRelease.title,
-    link: {
-      label: locale === 'it' ? 'Leggi' : 'Read',
-      href: LocalizeURL({
-        URL: '/press-releases/' + PRPage.slug,
-        locale,
-        defaultLocale,
-      }),
-    },
-    themeVariant: rest.themeVariant,
-  })),
-});
+  }): PressReleaseListProps => {
+  const resolvedLocale = locale || defaultLocale || 'en-GB';
+
+  return {
+    ...rest,
+    locale: resolvedLocale,
+    pressReleases: pressReleasePages.map((PRPage) => ({
+      date: formatDateToLocale(PRPage.pressRelease.date, resolvedLocale),
+      title: PRPage.pressRelease.title,
+      link: {
+        label: resolvedLocale.startsWith('it') ? 'Leggi' : 'Read',
+        href: LocalizeURL({
+          URL: '/press-releases/' + PRPage.slug,
+          locale: resolvedLocale,
+          defaultLocale,
+        }),
+      },
+      themeVariant: rest.themeVariant,
+      locale: resolvedLocale,
+    })),
+  };
+};
 
 const PressReleaseList = (
   props: PressReleaseListSection &
     SiteWidePageData & {
       pressReleasePages: ReadonlyArray<PressReleasePage>;
+      locale: string;
     }
-) => <PressReleaseListRC {...makePressReleaseListProps(props)} />;
+) => {
+  const { locale, defaultLocale } = props;
+  const resolvedLocale = locale || defaultLocale || 'en-GB';
+
+  return (
+    <PressReleaseListRC
+      {...makePressReleaseListProps({ ...props, locale: resolvedLocale })}
+    />
+  );
+};
```

Comment:
```suggestion
) => <PressReleaseListRC {...makePressReleaseListProps(props)} />;
```
No need to resolve anything.